tools/cjxl_main.cc

```diff
@@ -1153,7 +1153,8 @@ int main(int argc, char** argv) {
   params.runner = JxlThreadParallelRunner;
   params.runner_opaque = runner.get();
 
-  if (args.effort <= 6 || (args.streaming_input && args.streaming_output)) {
+  if ((args.effort < 7 || (args.effort == 7 && args.distance < 3)) ||
+      args.streaming_input) {
```

Comment:
Perhaps this logic belongs in the library itself so cjxl just uses "library defaults".

That also allows us to have more complicated defaults eventually, such as a heuristic that checks if the image would potentially benefit from patches/palette, and decides a suitable method accordingly.

Reply:
That's probably a valid point - @szabadka what do you think?

I would in general propose to do that as a follow-up though.

Reply:
Sounds good.